dadi/frontend/components/DocumentFilters/DocumentFilters.jsx

```diff
@@ -5,29 +5,70 @@ import {h, Component} from 'preact'
 import {isValidJSON} from 'lib/util'
 
 import DocumentFilter from 'components/DocumentFilter/DocumentFilter'
+import Button from 'components/Button/Button'
 
 export default class DocumentFilters extends Component {
+
+  componentWillMount() {
+    const {
+      filter, 
+      collection, 
+      field, 
+      value, 
+      fields
+    } = this.props
+
+    // Evaluate passed filter / store in state
+    this.state = {
+      edit: false,
+      visible: filter || false,
+      filters: isValidJSON(filter) ? JSON.parse(filter) : null
+    }
+  }
   render() {
-    const { filter, collection } = this.props
-    let filters = isValidJSON(filter) ? JSON.parse(this.props.filter) : null
-    
+    const {visible, filters, edit} = this.state
+    const {collection} = this.props
     return (
       <div>
-        {filters && Object.keys(filters).length && collection && (
+        <Button onClick={this.toggleFilters.bind(this)}>Filters</Button>
+
+        {visible && filters && Object.keys(filters).length && collection && (
           <div>
             {Object.keys(filters).map(key => ( 
-              <DocumentFilter field={key} value={filters[key]} fields={collection.fields} />
+              <DocumentFilter field={key} value={filters[key]} fields={collection.fields} updateFilter={this.updateUrlFilters.bind(this)} />
             ))}
           </div>
         )}
-        <hr />
-        <DocumentFilter fields={collection.fields} />
-        <button onClick={this.addFilter}>Add filter</button>
+        {edit && (
+          <DocumentFilter fields={collection.fields} updateFilter={this.updateUrlFilters.bind(this)} />
+        )}
+        {visible && (
+          <Button onClick={this.addFilter.bind(this)}>Add</Button>
+        )}
       </div>
     )
   }
 
+  updateUrlFilters(filter) {
+    const {updateUrlParams} = this.props
+    const {filters} = this.state
+
+    this.setState({filters: Object.assign(filters || {}, filter)})
```

Comment:
If `filters` exists and you do `Object.assign(filters, something)`, you're assigning something directly to a state variable, which we shouldn't do.

Why not just:

```js
this.setState({filters: Object.assign({}, filters, filter)})
```

It'll be fine even if `filters` is `null`.